src/main/java/org/jenkinsci/plugins/workflow/support/pickles/ExecutorPickle.java

```diff
@@ -88,6 +103,33 @@ protected Executor tryResolve() throws Exception {
                 // Or can we schedule a placeholder Task whose Executable does nothing but return Executor.currentExecutor and then end?
                 throw new IllegalStateException(exec + " was scheduled but no executor claimed it");
             }
+            @Override protected FlowExecutionOwner getOwner() {
+                return owner;
+            }
+            @Override protected void printWaitingMessage(TaskListener listener) {
+                Queue.Item item = Queue.getInstance().getItem(itemID);
+                String message = Messages.ExecutorPickle_waiting_to_resume(task.getFullDisplayName());
+                if (item == null) { // ???
+                    listener.getLogger().println(message);
+                    return;
+                }
+                CauseOfBlockage causeOfBlockage = item.getCauseOfBlockage();
+                if (causeOfBlockage != null) {
+                    listener.getLogger().print(message + ": ");
+                    causeOfBlockage.print(listener); // note that in case of Messages.Queue_Unknown for WaitingItem this is not very helpful
+                } else {
+                    listener.getLogger().println(message);
+                }
+            }
+            @Override public boolean cancel(boolean mayInterruptIfRunning) {
+                Queue.Item item = Queue.getInstance().getItem(itemID);
+                if (item != null) {
+                    if (!Queue.getInstance().cancel(item)) {
+                        LOGGER.log(Level.WARNING, "failed to cancel {0}", item);
+                    }
+                }
+                return super.cancel(mayInterruptIfRunning);
```

Comment:
Which part? The `Queue.cancel` call removes the _part of …_ task from the queue, which the last line of `unloadableExecutorPickle` asserts. The `super.cancel` stops the retrying in `TryRepeatedly` (though the next try would probably fail anyway, in `QueueTaskFuture.get`).